### PR TITLE
Tl.server

### PR DESCRIPTION
**Pull request by mysterytony**
## Summary

Add socket programming from express.cpp to server.cpp

## Related Issue

#9

### CODE REVIEW

File: express/express/server.cpp [hunk 1]
```diff
@@ -1,13 +1,143 @@
 #include "server.h"
+#include <sstream>
+
+using namespace std;
```
Comment (joshyuan360) [inline express/express/server.cpp]: use CExpress namespace OR remove std from code

Comment (joshyuan360) [inline express/express/server.cpp]: use CExpress namespace OR remove std from code

File: express/express/server.cpp [hunk 2]
```diff
@@ -1,13 +1,143 @@
...
+void response(SOCKET ClientSocket, std::function<std::string(std::string)> & processor) {
+
+  const int DEFAULT_BUFLEN = 2048;
+
+  char recvbuf[DEFAULT_BUFLEN];
+  int iResult, iSendResult;
+  int recvbuflen = DEFAULT_BUFLEN;
+
+  do {
+    // receive until the peer shuts down the connection
```
Comment (joshyuan360) [inline express/express/server.cpp]: use while (true)

File: express/express/server.h [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
-	class Server {
-	public:
-		Server();
-		void setPort(int);
-		void process(std::function<std::string (std::string)>);
-	};
-}
+  class Server {
+  public:
+    Server();
```
Comment (joshyuan360) [inline express/express/server.h]: include block comment for each public func

File: express/express/server.h [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
+#define WIN32_LEAN_AND_MEAN
+#endif
+
 #include <string>
-#include <memory>
+#include <windows.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#include <iphlpapi.h>
+#include <stdio.h>
```
Comment (joshyuan360) [inline express/express/server.h]: this one can be removed

File: express/express/server.h [hunk 5]
```diff
@@ -1,15 +1,31 @@
...
+    /* set port to server and open tcp ip*/
+    void setPort(int);
+    /*
+    listen on port and use the given lambda to process
+    the incoming connection and bytes
+    */
+    void process(std::function<std::string(std::string)>);
 
+  private:
+    WSADATA wsaData;
```
Comment (Xadoy) [inline express/express/server.h]: please refer to wiki for code style guide. The name of member variables should have appending underscore

File: express/express/server.h [hunk 6]
```diff
@@ -1,15 +1,31 @@
...
-		void setPort(int);
-		void process(std::function<std::string (std::string)>);
-	};
-}
+  class Server {
+  public:
+    /* default constructor*/
+    Server();
+    /* set port to server and open tcp ip*/
+    void setPort(int);
```
Comment (Xadoy) [inline express/express/server.h]: personally prefer set_port()

Comment (mysterytony) [inline express/express/server.h]: this is not a setter

**Pull request merged**